02-oop-inheritance/topic-2.js

```diff
@@ -0,0 +1,166 @@
+$(document).ready(function(){
+	/*Define Movie*/
+var Movie = function () {
+  /*Set Vars*/
+  this.attributes = {};
+  this.attributes.actors = [];
+};
+
+
+
+ /*Define Movies Prototype as a Module*/ 
+Movie.prototype = (function () {
+  var observers = [];
+  /*Public*/
+  return {
+    play: function () {
+      console.log('Movie/play');
+      this.notify('onPlay', this.attributes['title']);
+    },
+    stop: function () {
+      console.log('Movie/stop');
+      this.notify('onStop');
+    },
+    /*Get Method*/
+    get: function (attr) {
+      console.log('Movie/get');
+      return this.attributes[attr];
+    },
+    /*Set Method*/
+    set: function (attr, value) {
+      console.log('Movie/set');
+      this.attributes[attr] = value;
+    },
+    /*Observer Methods*/
+    addObserver: function (name, func) {
+      console.log('Movie/addObserver');
+      observers.push({name: name, func: func});
+    },
+    removeObserver: function (observer) {
+      console.log('Movie/removeObserver');
+      observers.slice(observer);
+    },
+    notify: function (obs, param) {
+      console.log('Movie/notify');
+      observers.forEach(function(observer) {
+        if (obs === observer.name) {
+         return observer.func(param);
+        }
+      });
+    },
+    /*Add an Actor*/
+    addAnActor: function(actor) {
+      console.log('Movie/setActors');
+      this.attributes.actors.push(actor);
+    },
+    /*Get Actors*/
+    getActors: function() {
+      console.log('Movie/getActors');
+      this.attributes.actors.forEach(function(actor){
+        console.log(actor.get('name'));
+      });
+    }
+  };
+})();
+
+/*Add Observers*/
+Movie.prototype.addObserver('onPlay', function(title) {
+  console.log('Playing ' + title + ' ...');
+  /*console.log(Movie.attributes);*/
+});
+
+Movie.prototype.addObserver('onStop', function() {
+  console.log('Stoped');
+});
+
+
+/*Create Instance of Movie*/
+var terminator = new Movie();
+terminator.set('title', 'Terminator');
+terminator.play();
+
+
+ /*Extend Method for Heritage*/
+extend = function(destination, source) {
+      for (var property in source) {
+        destination[property] = source[property];
+      }
+      return destination;
+};
+
+/*Define DownloadMovie Module*/
+var DownloadMovie = (function(){
```

Comment:
An IIFE is definitely not useful here, you could just have DownloadMovie being the returned object.